napari/utils/io.py

```diff
@@ -48,3 +48,27 @@ def imsave(filename: str, data: np.ndarray):
         import imageio
 
         imageio.imsave(filename, data)
+
+
+def __getattr__(name: str):
+    if name in {
+        'imsave_extensions',
+        'write_csv',
+        'read_csv',
+        'csv_to_layer_data',
+        'read_zarr_dataset',
+    }:
+        warnings.warn(
+            trans._(
+                '{name} was moved from napari.utils.io in 0.4.17. Import it from napari_builtins.io instead.',
+                deferred=True,
+                name=name,
+            ),
+            DeprecationWarning,
```

Comment:
```suggestion
            FutureWarning,
```
I know that it should be `DeprecationWarning`, but based on collected feedback, many plugin developers do not enable `DeprecationWarnings` and do not read warnings emitted during tests.

Reply:
I don't have strong feelings here and there doesn't seem to be a strong consensus about which one to use and when in the napari codebase. I mostly just want developers to be clearly informed of the change, so I'm happy to bring your suggestion in.